### PR TITLE
add std.getenv()

### PR DESCRIPTION
**Pull request by nigoroll**

### CODE REVIEW

File: lib/libvmod_std/vmod_std.c [hunk 1]
```diff
@@ -248,3 +248,12 @@ vmod_strstr(VRT_CTX, VCL_STRING s1, VCL_STRING s2)
...
 	return (strstr(s1, s2));
 }
+
+VCL_STRING __match_proto__(td_std_getenv)
+vmod_getenv(VRT_CTX, VCL_STRING name)
+{
+	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
+	if (name == NULL || *name == '\0')
+		return (NULL);
+	return (getenv(name));
```
Comment (Dridi) [inline lib/libvmod_std/vmod_std.c]: This should be copied in the workspace.

Comment (nigoroll) [inline lib/libvmod_std/vmod_std.c]: why?

Comment (Dridi) [inline lib/libvmod_std/vmod_std.c]: The prototype is `char *getenv(const char *name);`, it's not safe to assume it won't change during the transaction. A VMOD/library call may call `setenv` too.

Comment (nigoroll) [inline lib/libvmod_std/vmod_std.c]: it's not even save to assume that it won't change during the memcpy. So why bother?

Comment (Dridi) [inline lib/libvmod_std/vmod_std.c]: Transactions last orders of magnitude longer than a `memcpy`, but I agree it wouldn't be 100% safe either.

**Comment by bsdphk** [comment]
Dridi is right in that you cannot trust the constancy of the getenv(3)'s return if the process changes its own environment.

However, anybody who changes the environment of varnishd while it runs, deserve what they get as a result

OK

**Comment by nigoroll** [comment]
69838a5cc3250b79ad03d264ba49f37f814c6f8f

**Comment by hermunn** [comment]
Backport review: Backported as 503044881785

**Pull request closed without merging**